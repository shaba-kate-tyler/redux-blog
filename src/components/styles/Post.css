.Post {
    font-family: 'Roboto', sans-serif;
    padding: 1px;
    margin: 2px;
    a {
        text-decoration: none;
    }
    h2 {
        color: rgb(66,139,202);
    }
    p {
        font-weight: 100;
    }
}