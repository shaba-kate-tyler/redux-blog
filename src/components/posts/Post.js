import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import styles from '../styles/Post.css';

function Post({ title, body, id }) {
  return (
    <div className={styles.Post}>
      <Link to={`/posts/${id}`}>
        <h2>{title}</h2>
      </Link>
      <p>{body}</p>
    </div>
  );
}

Post.propTypes = {
  title: PropTypes.string.isRequired,
  body: PropTypes.string.isRequired,
  id: PropTypes.number.isRequired
};

export default Post;
