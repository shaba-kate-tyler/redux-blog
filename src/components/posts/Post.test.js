import React from 'react';
import renderer from 'react-test-renderer';
import Post from './Post';
import { MemoryRouter } from 'react-router-dom';

describe('Post', () => {
  it('matches a post snapshot', () => {
    const tree = renderer.create(
      <MemoryRouter>
        <Post
          title={'A Post'}
          body={'This is a post on a rant'}
          id={2}
        />
      </MemoryRouter>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
