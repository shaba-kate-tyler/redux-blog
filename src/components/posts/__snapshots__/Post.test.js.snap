// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Post matches a post snapshot 1`] = `
<div
  className="Post"
>
  <a
    href="/posts/2"
    onClick={[Function]}
  >
    <h2>
      A Post
    </h2>
  </a>
  <p>
    This is a post on a rant
  </p>
</div>
`;
